src/clientSideScene/sceneEntities.ts

```diff
@@ -1053,27 +1071,46 @@ export class SceneEntities {
             segmentName === 'tangentialArcTo'
           ) {
             resolvedFunctionName = 'tangentialArcTo'
+          } else if (snappedToTangent) {
+            // Generate tag for previous arc segment and use it for the angle of angledLine:
+            //   |> tangentialArcTo([5, -10], %, $arc001)
+            //   |> angledLine({ angle = tangentToEnd(arc001), length = 12 }, %)
+
+            const previousSegmentPathToNode = getNodePathFromSourceRange(
+              modifiedAst,
+              sourceRangeFromRust(lastSegment.__geoMeta.sourceRange)
+            )
+            const taggedAstResult = mutateAstWithTagForSketchSegment(
+              modifiedAst,
+              previousSegmentPathToNode
+            )
+            if (trap(taggedAstResult)) return Promise.reject(taggedAstResult)
+
+            modifiedAst = taggedAstResult.modifiedAst
+            snaps.previousArcTag = taggedAstResult.tag
+            resolvedFunctionName = 'angledLine'
           } else if (isHorizontal) {
             // If the angle between is 0 or 180 degrees (+/- the snapping angle), make the line an xLine
             resolvedFunctionName = 'xLine'
           } else if (isVertical) {
             // If the angle between is 90 or 270 degrees (+/- the snapping angle), make the line a yLine
             resolvedFunctionName = 'yLine'
-          } else if (snappedPoint.x === 0 || snappedPoint.y === 0) {
+          } else if (snappedPoint[0] === 0 || snappedPoint[1] === 0) {
             // We consider a point placed on axes or origin to be absolute
             resolvedFunctionName = 'lineTo'
           }
 
           const tmp = addNewSketchLn({
-            node: this.kclManager.ast,
+            node: modifiedAst,
             variables: this.kclManager.variables,
             input: {
               type: 'straight-segment',
               from: [lastSegment.to[0], lastSegment.to[1]],
-              to: [snappedPoint.x, snappedPoint.y],
+              to: [snappedPoint[0], snappedPoint[1]],
```

Comment:
Totally a nit, and I might be missing if `snappedPoint` is 3D:
```suggestion
              to: snappedPoint,
```

Reply:
It's not 3D!
The only reason I'm cloning the point when passing it to other functions is to avoid hard-to-debug bugs in case the called function mutates the passed argument which would still be used after the call in the caller function.
eg.:
```typescript
const snappedPoint = this.getSnappedDragPoint()
this.logSnapping(snappedPoint); // let's say this imaginary function modifies snappedPoint by mistake
// using snappedPoint here would lead to bugs
```

Cloning snappedPoint before passing it to `logSnapping` ensures that doesn't happen.
The bug is arguably in logSnapping in this case, but when it happens it is difficult to debug..

I think if it's safe to assume there is no modification it's better to go with your suggestion, it's cleaner code to the eyes as well. In this case I'm not sure because that input is being passed around a lot after the call, and the line above this ones suggests there was a reason for the cloning in the first place:
```typescript
from: [lastSegment.to[0], lastSegment.to[1]],
```